### DISCUSSION TITLE
acsses to Permission storge in android

### DISCUSSION THREAD

**Discussion by abdokhalil07** [flet-dev/flet#3068] [Q&A]
### Question

i want to save image 
 how can get permissin storge to save my file 
it work when i run in laptop(windows) but when
i build apk and run app and try to save image nothing save to my phone

### Code sample

```python
def link_generate_fun(e:FilePickerResultEvent):
        pasth = e.path
        #print(pasth)
        data = {
            'url':link_view.controls[0].controls[0].value,
            'name': link_view.controls[0].controls[1].value,
            }
        
        img = qrcode.make(data['url'])
        img.save(f'{pasth}.png')
        
        page.update()
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by suacorporate** [comment]
Hello, I also have the same issue; I can't get the application to prompt me for permissions.

**Comment by NHLOCAL** [comment]
I have the same problem. Also in the new version 0.23.1

**Comment by jtoror** [comment]
Hello, there are 2 ways, the bad and the ugly, the bad way is using p4a and modifying the AndroidManifest.xml and the ugly way is importing kivy and calling the platfrom function to request permissions and creating the apk with buildozier.

**Comment by ndonkoHenri** [comment]
I added the missing storage-permissions.
Could you guys give it another try with the latest Flet version.

**Reply by abdokhalil07** [reply]
ok its works in windows correctly, but in android get error " Unknown control : permission_handler"
![permission_erroe](https://github.com/flet-dev/flet/assets/94837888/756ede22-6286-4a5d-bbbf-875c7a63f897)

code : 

`from flet import *

def main(page: Page):
    page.scroll = ScrollMode.ADAPTIVE
    page.appbar = AppBar(title=Text("PermissionHandler Tests"))
    ph = PermissionHandler()
    page.overlay.append(ph)

    def check_permission(e):
        o = ph.check_permission(e.control.data)
        page.add(Text(f"Checked {e.control.data.name}: {o}"))

    def request_permission(e):
        o = ph.request_permission(e.control.data)
        page.add(Text(f"Requested {e.control.data.name}: {o}"))



    page.add(
        OutlinedButton(
            "Check Storage Permission",
            data=PermissionType.STORAGE,
            on_click=check_permission,
        ),
        OutlinedButton(
            "Request Microphone Permission",
            data=PermissionType.STORAGE,
            on_click=request_permission,

    ))

app(main)`

**Reply by ndonkoHenri** [reply]
Is that the Flet mobile app? (if yes, then sub to issue #3096)
Or an apk you built?

**Reply by abdokhalil07** [reply]
Flet mobile app
